### PR TITLE
Feature 176961984/presentation of fm v2 jobs

### PR DESCRIPTION
**Pull request by jkongie**
This PR has been separated into two commits, one for the API changes and one for the Operator UI. This API change introduces a breaking change to the jobs JSONAPI response which is why this PR should be reviewed together.

**API**
* Preload the Flux Monitor Spec when listing v2 jobs
* Preload the Flux Monitor Spec when showing v2 jobs
* Implement a Job presenter to represent the JSONAPI resource
* Update the jobs controller to use the presenter

**Operator UI**
* Update the list page to show flux monitor v2 jobs in the same style as v1 jobs
* Update the show job page to handle flux monitor v2 jobs

### CODE REVIEW

File: operator_ui/package.json
Comment (DeividasK) [inline operator_ui/package.json:102]: Prettier is already a repo level dep

Comment (jkongie) [reply operator_ui/package.json]: Removed

**Review by connorwstein — DISMISSED** [review]
Backend portion lgtm. I'll defer to @DeividasK for front end approval

File: operator_ui/src/pages/Jobs/generateJobSpecDefinition.ts
Comment (DeividasK) [inline operator_ui/src/pages/Jobs/generateJobSpecDefinition.ts:159]: Spreading attrs might be more succinct (unless there attrs we don't want)

File: operator_ui/src/pages/JobsIndex/DirectRequestRow.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/DirectRequestRow.tsx:73]: Named export. I haven't had a chance to, but if you have the time it would be great to add https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-default-export.md so comments like this would be redundant.

File: operator_ui/src/pages/JobsIndex/JobV2Row.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobV2Row.tsx:29]: Should be extracted out and shared between rows.

File: operator_ui/src/pages/JobsIndex/JobV2Row.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobV2Row.tsx:41]:
```suggestion
        return fluxMonitorSpec?.createdAt
```

Also, shouldn't need to do `.?` with a type update mentioned above

File: operator_ui/src/pages/JobsIndex/JobV2Row.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobV2Row.tsx:45]: ditto

File: operator_ui/src/pages/JobsIndex/JobV2Row.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobV2Row.tsx:93]: Can `createdAt` actually be undefined or is it just because of TS?

File: operator_ui/src/pages/JobsIndex/JobsIndex.test.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobsIndex.test.tsx:17]:
```suggestion
import { ENDPOINT as OCR_ENDPOINT } from 'api/v2/jobs'
```

File: operator_ui/src/pages/JobsIndex/JobsIndex.test.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobsIndex.test.tsx:9]:
```suggestion
} from 'factories/jsonApiJobs'
```

File: operator_ui/src/pages/JobsIndex/JobsIndex.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobsIndex.tsx:30]: Named imports

File: operator_ui/src/pages/JobsIndex/JobsIndex.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobsIndex.tsx:71]: Same Q about `createdAt` being undefined - is it possible in practice or just TS?

**Review by DeividasK — COMMENTED** [review]
No hard-blockers, but some inconsistencies + potential improvements.

File: operator_ui/src/pages/Jobs/Show.tsx
Comment (jkongie) [reply operator_ui/src/pages/Jobs/Show.tsx:98]: I believe this is resolved in that it removes the optional chaining however I believe we still need the switch, because the object just represents the API response, and you need to know the type to get the `createdAt` from the correct spec

File: operator_ui/src/pages/Jobs/generateJobSpecDefinition.ts
Comment (jkongie) [reply operator_ui/src/pages/Jobs/generateJobSpecDefinition.ts:159]: I generally like to be more explicit when the data comes from an external source (in this case it's the API) so that if the API changes, it doesn't, unbeknownst to us, add additional attributes. 

If you want this changed I could do this 

```
  const {
    fluxMonitorSpec,
    pipelineSpec,
    ...jobAttrs
  } = attrs
```

File: operator_ui/src/pages/JobsIndex/DirectRequestRow.tsx
Comment (jkongie) [reply operator_ui/src/pages/JobsIndex/DirectRequestRow.tsx:73]: OK I can add a ticket and do this in a separate commit

File: operator_ui/src/pages/JobsIndex/JobV2Row.tsx
Comment (jkongie) [reply operator_ui/src/pages/JobsIndex/JobV2Row.tsx:93]: It can't actually be undefined, however we have to get the `createdAt` attribute from the nested Spec in the Job, which we do through a switch.

```
  const createdAt = React.useMemo(() => {
    switch (job.attributes.type) {
      case 'fluxmonitor':
        return job.attributes.fluxMonitorSpec.createdAt
      case 'offchainreporting':
        return job.attributes.offChainReportingOracleSpec.createdAt
      default:
        return undefined
    }
  }, [job])
```

**Comment by jkongie** [comment]
@DeividasK Hey can you take another look at this please. I've made the changes you have requested.

Thanks!

File: operator_ui/@types/core/store/models.d.ts
Comment (DeividasK) [inline operator_ui/@types/core/store/models.d.ts:556]: No comments

File: operator_ui/src/pages/JobsIndex/JobsIndex.tsx
Comment (DeividasK) [inline operator_ui/src/pages/JobsIndex/JobsIndex.tsx:77]: Seems like https://github.com/smartcontractkit/chainlink/pull/3981/files#diff-d1ba52d002bf42ed9cc5b63f76ee3eab3352ed574832f6730db79b9da12c823aR14. If `createdAt` doesn't get brought up to the job level (which I think it should) this is gonna be a common op so might be worth creating v2 job utils.

I'd still prefer `createdAt` to live on `job`.

**Review by DeividasK — APPROVED** [review]
A few more nice-to-have comments, but overall LGTM 👍

File: operator_ui/src/pages/Jobs/Show.tsx
Comment (DeividasK) [reply operator_ui/src/pages/Jobs/Show.tsx]: I think this would be a better way to do this https://github.com/smartcontractkit/chainlink/pull/3981#discussion_r588100217

**Comment by jkongie** [comment]
Ok I'll get this merged now, because JB needs some changes in it

I'll address those issues in a separate PR

**Pull request merged**
